tests/datasource/fluent/test_sql_datasources.py

```diff
@@ -28,5 +31,143 @@ def test_kwargs_are_passed_to_create_engine(mocker: MockerFixture):
     )
 
 
+@pytest.mark.unit
+def test_table_quoted_name_type_does_not_exist(
+    mocker,
+):
+    """
+    DBMS entity names (table, column, etc.) must adhere to correct case insensitivity standards.  All upper case is
+    standard for Oracle, DB2, and Snowflake, while all lowercase is standard for SQLAlchemy; hence, proper conversion to
+    quoted names must occur.  This test ensures that mechanism for detection of non-existent table_nam" works correctly.
+    """
+    table_names_in_dbms_schema: list[str] = [
+        "table_name_0",
+        "table_name_1",
+        "table_name_2",
+        "table_name_3",
+    ]
+
+    with mock.patch(
+        "great_expectations.datasource.fluent.sql_datasource.TableAsset.datasource",
+        new_callable=mock.PropertyMock,
+        return_value=SQLDatasource(
+            name="my_snowflake_datasource",
+            connection_string="snowflake://<user_login_name>:<password>@<account_identifier>/<database_name>/<schema_name>?warehouse=<warehouse_name>&role=<role_name>",
+        ),
+    ):
+        table_asset = TableAsset(
+            name="my_table_asset",
+            table_name="nonexistent_table_name",
+            schema_name="my_schema",
+        )
+        assert table_asset.table_name not in table_names_in_dbms_schema
+
+
+@pytest.mark.unit
+def test_table_quoted_name_type_all_upper_case_normalizion_is_noop():
+    """
+    DBMS entity names (table, column, etc.) must adhere to correct case insensitivity standards.  All upper case is
+    standard for Oracle, DB2, and Snowflake, while all lowercase is standard for SQLAlchemy; hence, proper conversion to
+    quoted names must occur.  This test ensures that all upper case entity usage does not undergo any conversion.
+    """
+    table_names_in_dbms_schema: list[str] = [
```

Comment:
Just a small item: We only need one or maybe two (with, without underscore) names for this test, no need to test the whole list.

Reply:
@anthonyburdi Should I make it smaller, or leave as is (nice names :-) there)?

Reply:
@anthonyburdi This has been committed -- thanks.